protocol/validation/block.go

```diff
@@ -32,7 +32,7 @@ var (
 // See $CHAIN/protocol/doc/spec/validation.md#accept-block.
 // It evaluates the prevBlock's consensus program,
 // then calls ValidateBlock.
-func ValidateBlockForAccept(ctx context.Context, snapshot *state.Snapshot, prevBlock, block *bc.Block, validateTx func(*bc.Tx) error) error {
+func ValidateBlockForAccept(ctx context.Context, snapshot *state.Snapshot, prevBlock, block *bc.Block, initialBlockHash bc.Hash, validateTx func(*bc.Tx) error) error {
```

Comment:
What do you think about putting the initial block hash in state.Snapshot? (Not in the serialized snapshot data, just the internal representation.)

I don't know how that would shake out exactly, but it might be a not-entirely-unreasonable place for it, since the state is tied to the initial block hash anyway.